File: models/src/main/java/ar/edu/itba/paw/models/Office.java
package ar.edu.itba.paw.models;

import ar.edu.itba.paw.persistenceAnnotations.Column;
import ar.edu.itba.paw.persistenceAnnotations.OrderBy;
import ar.edu.itba.paw.persistenceAnnotations.OrderCriteria;
import ar.edu.itba.paw.persistenceAnnotations.Table;

@Table(name = "office", primaryKey = "office_id")
public class Office extends GenericModel<Integer> {
    @Column(name = "phone")
    private String phone;
    @Column(name = "email")
    private String email;
    @OrderBy(OrderCriteria.ASC)
    @Column(name = "name", required = true)
    private String name;
    @Column(name = "street", required = true)
    private String street;
    @Column(name = "url")
    private String url;

    private Locality locality;

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return this.phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return this.email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getStreet() {
        return this.street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public Locality getLocality() {
        return this.locality;
    }

    public void setLocality(Locality locality) {
        this.locality = locality;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    @Override
    protected boolean isSameInstance(Object o) {
        return o instanceof Office;
    }
}

File: vueapp/src/logic/models/Office.ts
import {GenericEntity} from '~/src/logic/models/utils/GenericEntity';
import {JSONSerializableKeys} from '~/src/logic/models/utils/JSONSerializable';
import {Locality} from '~/src/logic/models/Locality';

export class Office extends GenericEntity<Office> {
    private _id: number;
    private _name: string;
    private _phone: string;
    private _email: string;
    private _street: string;
    private _url: string;
    private _locality: Locality;

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get name(): string {
        return this._name;
    }

    public set name(value: string) {
        this._name = value;
    }

    public get phone(): string {
        return this._phone;
    }

    public set phone(value: string) {
        this._phone = value;
    }

    public get email(): string {
        return this._email;
    }

    public set email(value: string) {
        this._email = value;
    }

    public get street(): string {
        return this._street;
    }

    public set street(value: string) {
        this._street = value;
    }

    public get url(): string {
        return this._url;
    }

    public set url(value: string) {
        this._url = value;
    }

    public get locality(): Locality {
        return this._locality;
    }

    public set locality(value: Locality) {
        this._locality = value;
    }

    public toJSON(): JSONSerializableKeys<Office> {
        return {
            id: this.id,
            email: this.email,
            locality: this.locality,
            name: this.name,
            phone: this.phone,
            street: this.street,
            url: this.url
        };
    }
}
